content/ko/docs/tasks/job/coarse-parallel-processing-work-queue.md

```diff
@@ -0,0 +1,326 @@
+
+## 잡 실행
+
+이제 잡을 실행한다.
+
+```shell
+kubectl apply -f ./job.yaml
+```
+
+이제 조금 기다린 다음, 잡을 확인한다.
+
+```shell
+kubectl describe jobs/job-wq-1
+```
+```
+Name:             job-wq-1
+Namespace:        default
+Selector:         controller-uid=41d75705-92df-11e7-b85e-fa163ee3c11f
+Labels:           controller-uid=41d75705-92df-11e7-b85e-fa163ee3c11f
+                  job-name=job-wq-1
+Annotations:      <none>
+Parallelism:      2
+Completions:      8
+Start Time:       Wed, 06 Sep 2017 16:42:02 +0800
+Pods Statuses:    0 Running / 8 Succeeded / 0 Failed
+Pod Template:
+  Labels:       controller-uid=41d75705-92df-11e7-b85e-fa163ee3c11f
+                job-name=job-wq-1
+  Containers:
+   c:
+    Image:      gcr.io/causal-jigsaw-637/job-wq-1
+    Port:
+    Environment:
+      BROKER_URL:       amqp://guest:guest@rabbitmq-service:5672
+      QUEUE:            job1
+    Mounts:             <none>
+  Volumes:              <none>
+Events:
+  FirstSeen  LastSeen   Count    From    SubobjectPath    Type      Reason              Message
+  ─────────  ────────   ─────    ────    ─────────────    ──────    ──────              ───────
+  27s        27s        1        {job }                   Normal    SuccessfulCreate    Created pod: job-wq-1-hcobb
+  27s        27s        1        {job }                   Normal    SuccessfulCreate    Created pod: job-wq-1-weytj
+  27s        27s        1        {job }                   Normal    SuccessfulCreate    Created pod: job-wq-1-qaam5
+  27s        27s        1        {job }                   Normal    SuccessfulCreate    Created pod: job-wq-1-b67sr
+  26s        26s        1        {job }                   Normal    SuccessfulCreate    Created pod: job-wq-1-xe5hj
+  15s        15s        1        {job }                   Normal    SuccessfulCreate    Created pod: job-wq-1-w2zqe
+  14s        14s        1        {job }                   Normal    SuccessfulCreate    Created pod: job-wq-1-d6ppa
+  14s        14s        1        {job }                   Normal    SuccessfulCreate    Created pod: job-wq-1-p17e0
+```
+
+모든 파드가 성공했다. 야호.
+
+
+
+<!-- discussion -->
+
+## 대안
+
+이러한 접근은 "워커" 프로그램을 작업 대기열에 맞게 수정하지 않아도 
+된다는 장점이 있다.
+
+이 접근을 이용하려면, 메시지 큐 서비스를 실행해야만 한다.
+만약 메시지 큐 서비스를 실행하는 게 불편하다면, 
+다른 [잡 패턴](/docs/concepts/workloads/controllers/job/#job-patterns)을 고려해볼 수 있다.
+
+이 접근은 모든 작업 아이템에 대해 파드를 생성한다. 만약 작업 아이템이 오직 몇 초밖에 걸리지 않는 작업이라면, 
+매 작업마다 파드를 생성하는 것은 아주 큰 오버헤드를 더할 수 있다.
+하나의 파드가 여러 작업 아이템을 수행하는 이 [예제](/docs/tasks/job/fine-parallel-processing-work-queue/)를 고려해보자.
+
+이 예시에서는, `amqp-consume` 유틸리티를 이용해 큐로브터 메시지를 읽어 
+실제 프로그램을 실행했다. 이러면 메시지 큐를 이용하기 위해 프로그램을 수정하지 
+안항도 된다는 장점이 있다. 
+[다른 예제](/docs/tasks/job/fine-parallel-processing-work-queue/)는 
+클라이언트 라이브러리를 이용해 작업 큐와 소통하는 방법을 보여준다.
+
+
+## 주의 사항
+
+만약 작업 완료 수가 큐에 있는 아이템의 숫자보다 적게 설정되면, 
+모든 아이템 처리되지 않는다.
+만약 작업 완료 수가 큐에 있는 아이템의 숫자보다 많게 설정되면, 
+큐에 있는 아이템이 모두 처리되어도, 잡이 완료됐다고 표시되지 않고, 
+메시지를 기다리는 파드를 추가적으로 실행시킨다.
```

Comment:
꼼꼼히 리뷰해주셔서 정말 감사합니다!

> It will start additional pods which will block waiting for a message.

에서 'block waiting'이 의미하는 바가 [kubeadm blocks waiting for control plane during installation](https://kubernetes.io/docs/setup/production-environment/tools/kubeadm/troubleshooting-kubeadm/#kubeadm-blocks-waiting-for-control-plane-during-installation) 에서 kubeadm이 작업을 계속하지 못하고 대기하는 것을 표현했듯이, 대기열에 더 작업할 아이템이 없어, 파드가 대기열의 메시지를 기다리며 멈춰있는 (block) 상태를 의미한다고 생각했습니다. 

그래서 작업 완료 수를 맞추기 위해서 우선 파드를 만들지만, 대기열에서 매시지를 받아오지 못하기 때문에 이 새로운 파드들이 메시지를 기다리며 계속 대기하게 된다는 뜻으로 해석했습니다.

같은 의미에서 '차단'이라는 표현을 쓸 수도 있을 것 같습니다. 하지만 '메시지 대기를 차단하는 파드'라고 하면 메시지 대기를 차단하는 목적으로 새로운 파드를 만든다는 오해를 만들 수 있을 것 같습니다. 

```
만약 작업 완료 수가 큐에 있는 아이템의 숫자보다 많게 설정되면,
대기열에 있는 아이템이 모두 처리되어도, 잡이 완료됐다고 표시되지
않고, 메시지를 기다리는 과정에서 막히는 파드를
추가적으로 실행시킨다.
```
는 어떨까요?

Reply:
네, 말씀대로 수정하는 것이 더 나을 것 같습니다. 👍

Reply:
수정해서 커밋했습니다! :)